src/Components/NewArrivals/NewArrivals.jsx

```diff
@@ -0,0 +1,23 @@
+import { useProductProvider } from "../../Context/ProductsProvider";
+import ProductsCard from "../ProductListingPage/ProductsSection/ProductsCard/ProductsCard";
+
+const NewArrivals = () => {
+  const { productState } = useProductProvider();
+  return (
+    <div className="similar-products">
+      <h1 className="title-lg-wt-5 mg-2-bot text-center">New Arrivals</h1>
+      <div className="flex-row-center">
+        {productState.products.map((item) => {
+          return item.newestArrival ? (
+            <ProductsCard key={item._id} item={item} />
+          ) : (
+            false
+          );
```

Comment:
Can you please explain why returning false? This logic seems inappropriate.

Reply:
Using ternary to check with the condition of the new arrivals, else would have to use logical && operator which is not good for some cases. 
I am following the practice to use ternary as per https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx article.

Reply:
Using ternary operator is okay but returning false is not. You should return **null** instead. Also at times, you can use ***&&*** operator. The issue with **&&** operator occurs when you do something like 
```array.length && Some Jsx```, as mentioned in the blog, it retuns **0** in the initial render. You can probably use **&&** operator here, as your code is not checking for **array.length**

Reply:
Oh ok, I didn't know that, I think I took it in the wrong direction. 
Thank you so much for the clarification, will try to optimize the code with && operator. 😊